app/controllers/ruleset_controller.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+class RulesetController < ApplicationController
+
+  before_action :set_ruleset_type, only: [:new, :create]
+
+  def new
+    render @ruleset_name
+  end
+
+  def create
+    if ruleset_params[:oas_file].nil?
+      flash[:error] = 'Please upload a file'
+      return render @ruleset_name
+    end
+
+    # Here we can call our 42 crunch or spectral service
+
+    render json: { message: 'File uploaded successfully' }
+  end
+
+  private
+
+  def set_ruleset_type
+    @ruleset_name = ruleset_params[:which_ruleset]
```

Comment:
In this one method, a single parameter is given three different labels:
- which_ruleset
- ruleset_name
- ruleset_type

To reduce this the method name should be changed to `set_ruleset_name`